### PR TITLE
Sanitize errors 

### PR DESCRIPTION
**Pull request by mattnaides**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
+                code: sanitizedErr.code,
+                signal: sanitizedErr.signal
+            }, sanitizedErr)
+        }
     }
 });
 
+function sanitizeObject(input: { [key: string]: any }) {
+    for (const k of Object.keys(input)) {
+        if (SANITIZED_KEYS.has(k)) {
```
Comment (scottbessler) [inline index.ts]: i wonder if we should make this a fuzzy match (anything that contains one of those 3 words)

File: index.ts [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
     serializers: {
-        err: err => Object.assign({
-            message: err.message,
-            name: err.name,
-            stack: getFullErrorStack(err),
-            code: err.code,
-            signal: err.signal
-        }, err)
+        err: err => {
+            const sanitizedErr = JSON.parse(JSON.stringify(err))
```
Comment (scottbessler) [inline index.ts]: hmm are we sure this works for errors?

Comment (scottbessler) [reply index.ts]:
```
JSON.stringify(new Error('foo'))
'{}'
```

File: index.ts [hunk 3]
```diff
@@ -1,21 +1,39 @@
...
+                name: err.name,
+                stack: getFullErrorStack(err),
+                code: err.code,
+                signal: err.signal
+            }, err)
+        }
     }
 });
 
+function sanitizeObject(input: { [key: string]: any }) {
```
Comment (scottbessler) [inline index.ts]: could we instead have sanitizeObject return a new object  that we assign onto the old one?
e.g. returns `{ stupid_password: 'SANITIZED' } `

Comment (scottbessler) [reply index.ts]: (that way we dont have to worry about including a non-error object ON an error as context)

Comment (mattnaides) [reply index.ts]: @scottbessler I'm not sure I understand your request. 
```
sanitizeObject({
   "secret": "my secret",
   "color": "blue",
   "nested" : {
      "my_cred" : "sanitize me",
      "loud" : true
   }
})

returns 
{
   "secret": "SANITIZED",
   "nested" : {
      "my_cred" : "SANITIZED",
      "loud" : true
   }
}
```
Which we'd then assign into the error like
```
err => {...err, ...sanitizeObject(err)}
```
```

Comment (scottbessler) [reply index.ts]: exactly except dont mutate the input in sanitizeobject

**Pull request closed without merging**